crypto/fipsmodule/bcm.c

```diff
@@ -174,7 +184,8 @@ static void BORINGSSL_maybe_set_module_text_permissions(int permission) {}
 
 static void __attribute__((constructor))
 BORINGSSL_bcm_power_on_self_test(void) {
-  CRYPTO_library_init();
+
+  OPENSSL_cpuid_setup();
```

Comment:
This is a change of behaviour. Consuming applications can call `CRYPTO_library_init()`, which, in turn, calls `OPENSSL_cpuid_setup()`, but only calls the latter function once.

Hence, if any of the functions `OPENSSL_cpuid_setup()` are not idempotent, (e.g. depends on external data) then the cpu capabilities might change after returning from `BORINGSSL_bcm_power_on_self_test()`.

Reply:
This should be fine though! Since `do_library_init()` no longer does anything in fips mode.
https://github.com/awslabs/aws-lc/blob/feae2ebf5c31e8fe357b80c69b2e4eccc896bfbb/crypto/crypto.c#L74